src/views/Profile/components/EditProfileModal/ApproveCakePage.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useState } from 'react'
+import { useWallet } from '@binance-chain/bsc-use-wallet'
+import { AutoRenewIcon, Button, Flex, InjectedModalProps, Text } from '@pancakeswap-libs/uikit'
+import useI18n from 'hooks/useI18n'
+import { useCake } from 'hooks/useContract'
+import { useProfile, useToast } from 'state/hooks'
+import { getPancakeProfileAddress } from 'utils/addressHelpers'
+import { getFullDisplayBalance } from 'utils/formatBalance'
+import useGetProfileCosts from '../../hooks/useGetProfileCosts'
+import { UseEditProfileResponse } from './reducer'
+
+interface ApproveCakePageProps extends InjectedModalProps {
+  goToChange: UseEditProfileResponse['goToChange']
+}
+
+const ApproveCakePage: React.FC<ApproveCakePageProps> = ({ goToChange, onDismiss }) => {
+  const [isApproving, setIsApproving] = useState(false)
+  const { profile } = useProfile()
+  const TranslateString = useI18n()
+  const { account } = useWallet()
+  const { numberCakeToUpdate, numberCakeToReactivate } = useGetProfileCosts()
+  const cakeContract = useCake()
+  const { toastError } = useToast()
+  const cost = profile.isActive ? numberCakeToUpdate : numberCakeToReactivate
+
+  const handleApprove = () => {
+    cakeContract.methods
+      .approve(getPancakeProfileAddress(), cost.times(2).toJSON())
```

Comment:
what happen if I create, pause, enable, pause, enable, ....
The allowance will probably be exceeded

Reply:
Ok I just saw that you check the allowance first, then it's ok